### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by sagittarius-a**
In order to ease the pull request process on both sides, a template
would help the user to quickly identify and track actions to perform.

Since there is an existing issue template, I thought that this would be convenient to have as well.

Structure is not very well defined. I do not have a strong experience with this project yet so I drafted a template based on my experience with PRs.

Feel free to discard or update sections.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+See [CONTRIBUTING.md User Documentation section](../CONTRIBUTING.md#user-documentation) for further details.
+
+**Note: Manual page changes must be performed in a commit, not in this PR section.**
+
+# Checklist:
+
+- [ ] Tests pass with `cargo test --all-targets --all-features`
+- [ ] Code is formatted with `cargo fmt -- --check`
+- [ ] Code is reviewed with `cargo clippy --release`
```
Comment (VuiMuich) [inline .github/pull_request_template.md:26]: Those three lines feel kind of redundant to the CI checks, though for completeness they also make sense.

Comment (VuiMuich) [inline .github/pull_request_template.md:26]: Those three lines feel kind of redundant to the CI checks, though for completeness they also make sense.

Comment (sagittarius-a) [reply .github/pull_request_template.md:26]: Absolutely. I tend to run these checks on my own to make sure no further commit will be required to fix non passing CI.

Comment (VuiMuich) [reply .github/pull_request_template.md:26]: Maybe we could condense this to 'ran `make test` lokalliy with no errors or warnings reported'

**Pull request merged**